pkg/nsm/apis/mechanism/mechanism.proto

```diff
@@ -0,0 +1,14 @@
+syntax = "proto3";
```

Comment:
Have a separate proto just to declare single message looks really odd. I do not get why?

Reply:
I totally agree that it looks odd.  Here's why.

If I put:

```proto
message Mechanism {
    oneof mechanism {
        LocalMechanism local = 1;
        RemoteMechanism remote = 2;
    }
}
```
in common.proto

and run
```bash
cd pkg;go generate ./...;cd -
GOPATH=${GOPATH} ./scripts/update-codegen.sh && ./scripts/add-openapi-bits.sh
cd pkg;go build ./...
```

I get an error:

```
nsm/apis/common/zz_generated.deepcopy.go:76:31: in.Mechanism.DeepCopyisMechanism_Mechanism undefined (type isMechanism_Mechanism has no field or method DeepCopyisMechanism_Mechanism)
```

Basically, the magic we use to generate the deepcopy needed for CRDs doesn't work with proto oneof.  So I *can't* define Mechanism in a proto file that is used to generate a CRD

Since LocalMechanism is being used in the NSE CRD, and LocalMechanism is defined in common... I can't use define it in common.

Reply:
Basically, the net-net is: the generated deep copies can't handle interfaces, and proto-gen uses interfaces with oneof...